gamersnet_frontend/src/components/displayPosts/index.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import {withRouter} from 'react-router-dom';
+
+import APIFetch from '../../utilities/api';
+import Post from './post'
+
+class DisplayPosts extends React.Component{
+    _isMounted = false;
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            items: []
+        };
+        this.delete_post = this.delete_post.bind(this);
+    }
+
+    delete_post(id){
+        this.setState(prevState => ({
+            items: prevState.items.filter(post => post["_id"] != id  )
+        }));  
+    }
+
+    componentDidMount() {
+        this._isMounted = true;
+
+        let json = {userID : ""};
+        let fetchPosts = APIFetch('/posts/listUserPosts', JSON.stringify(json),'POST');
+        // for peer reviewers, How can we do error checking here?
+        fetchPosts
+          .then(res => res.json())
+          .then((result) => {
+            console.log(result)
+            this.setState({items: result});
+        });
```

Comment:
Consider the following code excerpt from my `account_feature` branch (`frontend/src/components/profileInfo/`):
```javascript
componentDidMount() {
    let fetchData = APIFetch('/users/getUserDetails/' + this.props.userID);

    fetchData.then(async (data) => {
        if (await data.ok) {
            // if http status == 2xx
            let json = await data.json();
                
            this.props.sendToParent(json.details);

            this.setState({status: 1, data: json});
        } else if (await data.status === 404) {
            // if http status == 404 (user ID not found)
            this.setState({data: {username: `404, user ${this.props.userID} not found`}});
        } else {
            // catch all other http errors
            console.log('profile-info', 'network problem happened');
        }
    });
}
```

Reply:
I did this for other components but for some reason, async was not working. I'll take another look at it and fix all the changes.